resources/pip.rb

```diff
@@ -98,8 +98,10 @@ def require_install?
     end
 
     unless new_resource.version
-      Chef::Log.debug("already installed: #{new_resource.package_name} #{current_version}")
-      return false
+      unless new_resource.options.split(/\s+/).include? '--upgrade'
+        Chef::Log.debug("already installed: #{new_resource.package_name} #{current_version}")
+        return false
+      end
```

Comment:
Looking at this more closely, I feel a better approach might be to create a new `:upgrade` action which can properly deal with this logic a little better.